frappe/desk/form/linked_with.py

```diff
@@ -79,18 +79,19 @@ def get_submitted_linked_docs(doctype, name, docs=None, visited=None):
 @frappe.whitelist()
 def cancel_all_linked_docs(docs, ignore_doctypes_on_cancel_all=[]):
 	"""
-	Cancel all linked doctype
+	Cancel all linked doctype, optionally ignore doctypes specified in a list.
 
 	Arguments:
-		docs (str) - It contains all list of dictionaries of a linked documents.
+		docs (json str) - It contains list of dictionaries of a linked documents.
+		ignore_doctypes_on_cancel_all (list) - List of doctypes to ignore while cancelling.
 	"""
 
 	docs = json.loads(docs)
 	if isinstance(ignore_doctypes_on_cancel_all, string_types):
 		ignore_doctypes_on_cancel_all = json.loads(ignore_doctypes_on_cancel_all)
 	for i, doc in enumerate(docs, 1):
-		if validate_linked_doc(doc, ignore_doctypes_on_cancel_all) is True:
-			frappe.publish_progress(percent=i * 100 / ((len(docs) - len(ignore_doctypes_on_cancel_all))), title=_("Cancelling documents"))
+		if validate_linked_doc(doc, ignore_doctypes_on_cancel_all):
+			frappe.publish_progress(percent=i * 100 / len(docs), title=_("Cancelling documents"))
```

Comment:
Shouldn't this be outside `if` condition to get the actual progress?

Currently, it seems progress will never be 100% if any `doc` in `docs` list fails the validation (`validate_linked_doc`).

Reply:
Fixed